### DISCUSSION TITLE
Plots in tabs not displaying userscript topics for full time range on recorded bag

### DISCUSSION THREAD

**Discussion by rg-gravis** [foxglove/community#1028] [Bugs] [answered]
To replicate: using `example-001-av`, add the user script
```ts
import { Input, Message } from "./types";

type Output = {
  world_accel: number;
};

export const inputs = ["/imu"];
export const output = "/studio_script/output_topic";

export default function script(event: Input<"/imu">): Output {
  return {
    world_accel: Math.hypot(
      event.message.linear_accel!.x!,
      event.message.linear_accel!.y!,
      event.message.linear_accel!.z!,
    ),
  };
}
```
Then add `/studio_script/output_topic.world_accel` as a series to one of the plots in the "Controls" panel, switch to a different panel (e.g. "Planning"), and reload the data. If you load data and start playback on a different panel, then switch to the controls panel to see the plot, the `/studio_script/output_topic.world_accel` plot will only draw in as messages are played, instead of loading for the entire duration like the other series.
 Also, if you do this with a plot that has a "Path (accumulated)" x-axis (e.g. `/imu.linear_accel.x` as X, with `/studio_script/output_topic.world_accel` as a series), then it will display correctly if data is loaded with the "Controls" panel visible, but will give a "Mismatch between x- and y-axis data sizes" error if data is loaded on a different panel.
 
 I'm using the desktop app, on ubuntu 20.04, Foxglove Studio version 2.12.1

**Comment by defunctzombie** [comment]
@bryfox I think you are aware of this and working on a fix?

**Reply by bryfox** [reply]
Yes — I will confirm with the steps above, but this sounds like a bug I'm actively working on.

**Comment by bryfox** [comment, marked as answer]
A fix for this is rolling out now, and it'll be included in the next Desktop release (>2.12.1). I was able to reproduce the issue, and see the fix. If you still encounter problems, please re-open.
